src/accessibility/index.ts

```diff
@@ -0,0 +1,57 @@
+import type { ElementHandle } from 'puppeteer';
+import { createClientRuntimeServer } from '../module-server/client-runtime-server';
+import { assertElementHandle } from '../utils';
+
+const accessibilityTreeSymbol: unique symbol = Symbol('PT Accessibility Tree');
+
+export interface AccessibilityTreeOptions {
+  /**
+   * Whether the accessible description of elements should be included in the tree.
+   * https://www.w3.org/TR/wai-aria-1.2/#dfn-accessible-description
+   * (default: true)
+   */
+  includeDescriptions?: boolean;
+  /**
+   * Whether to include text that is not part of an element's accessible name/description
+   * (default: false)
+   */
+  includeText?: boolean;
+}
+
+export const getAccessibilityTree = async (
+  element: ElementHandle,
+  options: AccessibilityTreeOptions = {},
+) => {
+  const serverPromise = createClientRuntimeServer();
+  assertElementHandle(element, getAccessibilityTree);
+
+  const { port } = await serverPromise;
+
+  const result: string = await element.evaluate(
+    // Using new Function to avoid babel transpiling the import
+    // @ts-expect-error pptr's types don't like new Function
+    new Function(
+      'element',
+      'options',
+      `return import("http://localhost:${port}/@pleasantest/accessibility")
+        .then(accessibility => accessibility.getAccessibilityTree(element, options))`,
+    ),
+    options,
+  );
+
+  return {
+    [accessibilityTreeSymbol]: result,
+    toString: () => result,
+  };
+};
+
+expect.addSnapshotSerializer({
+  serialize: (val, config, indentation, depth, refs, printer) => {
```

Comment:
I’m assuming this lint warning needs an ignore comment?

Reply:
It is a warning so it doesn't fail the linting, it just appears as a warning in the eslint output.

I'll probably disable that rule for this project